controllers/storagecluster/odfinfoconfig.go

```diff
@@ -0,0 +1,251 @@
+package storagecluster
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v4/v1"
+	ocsv1a1 "github.com/red-hat-storage/ocs-operator/api/v4/v1alpha1"
+	"github.com/red-hat-storage/ocs-operator/v4/controllers/util"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/yaml"
+)
+
+type connectedClient struct {
+	Name      string `yaml:"name"`
+	ClusterID string `yaml:"clusterId"`
+}
+type infoStorageCluster struct {
+	NamespacedName          types.NamespacedName `yaml:"name"`
+	StorageProviderEndpoint string               `yaml:"storageProviderEndpoint"`
+	CephClusterFSID         string               `yaml:"cephClusterFSID"`
+}
+
+type OdfInfoData struct {
+	Version           string             `yaml:"version"`
+	DeploymentType    string             `yaml:"deploymentType"`
+	Clients           []connectedClient  `yaml:"clients"`
+	StorageCluster    infoStorageCluster `yaml:"storageCluster"`
+	StorageSystemName string             `yaml:"storageSystemName"`
+}
+
+const (
+	odfInfoKeySuffix          = "config.yaml"
+	odfDeploymentTypeExternal = "external"
+	odfDeploymentTypeInternal = "internal"
+	rookCephMonSecretName     = "rook-ceph-mon"
+	fsidKey                   = "fsid"
+	odfOperatorNamePrefix     = "odf-operator"
+	odfInfoConfigMapName      = "odf-info"
+	odfInfoMapKind            = "ConfigMap"
+)
+
+type odfInfoConfig struct{}
+
+var mutex sync.RWMutex
+
+// ensureCreated ensures that a ConfigMap resource exists with its Spec in
+// the desired state.
+func (obj *odfInfoConfig) ensureCreated(r *StorageClusterReconciler, storageCluster *ocsv1.StorageCluster) (reconcile.Result, error) {
+	operatorNamespace, err := util.GetOperatorNamespace()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	odfInfoConfigMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      odfInfoConfigMapName,
+			Namespace: operatorNamespace,
+		},
+	}
+
+	mutex.Lock()
+	defer mutex.Unlock()
+	_, err = ctrl.CreateOrUpdate(r.ctx, r.Client, odfInfoConfigMap, func() error {
+		if err = ctrl.SetControllerReference(storageCluster, odfInfoConfigMap, r.Scheme); err != nil {
+			return err
+		}
+		r.Log.Info("Creating or updating odf-info config map", odfInfoMapKind, klog.KRef(storageCluster.Namespace, odfInfoConfigMapName))
```

Comment:
We have `client.ObjectKeyFromKey` which takes the object itself and produces a namespace name to be logged. 
There is no need to introduce a new way of creating namespace names where controller runtime already provides a native way to achieve the same 

```suggestion
		r.Log.Info("Creating or updating odf-info config map", odfInfoMapKind, klog.KRef(storageCluster.Namespace, odfInfoConfigMapName))
```

Reply:
[Here](https://github.com/red-hat-storage/ocs-operator/pull/2440#discussion_r1584832378)'s where I commented regarding this exact same matter. I thought surely there's a significance to the usage of `klog` versus w/o it.

Reply:
I am not aware of any significant meaning behind using klog. To me this seemed like another unnecessary concept the developer would have to keep in mind when coming to contribute future changes to this code